tests/nn/data_parallel/test_sharded_ddp.py

```diff
@@ -52,7 +59,8 @@ def run_one_step(rank, world_size, backend, device, temp_file_name):
     optimizer = ddp.optimizer
     model = ddp.module
 
-    input_tensor = torch.rand((64, 2)).to(device)
+    # Different input per rank, allows for checking that the gradients have been properly reduced
+    input_tensor = (torch.ones((64, 2)) * rank).to(device)
```

Comment:
this brings the inputs on one rank to 0, which kills one of the gradients (unless it has been properly synced), which makes it a better unit test.

Reply:
I'm not sure this adjustment would have caught the earlier bug. Could we check the reduced gradient values are what we expect?

Reply:
What is the purpose of the `.abs().sum() / input_tensor.numel()` below ?

Reply:
this means that the gradient on the weights is zero on the first layer on one of the ranks, so that if the grads are not reduced it is caught. Commenting out the ddp.reduce() or changing the ddp code to work on params instead of gradients does break this test. Now if you want another test, anything is possible of course